compiler-core/src/error.rs

```diff
@@ -149,7 +151,11 @@ pub enum Error {
     Gzip(String),
 
     #[error("shell program `{program}` not found")]
-    ShellProgramNotFound { program: String },
+    ShellProgramNotFound {
+        program: String,
+        os: String,
+        distro: String,
```

Comment:
Let's use enum types for these rather than strings.

Reply:
ok, I added the enums in `compiler-core/src/error.rs`